src/converse-core.js

```diff
@@ -4893,7 +4893,12 @@
                         throw new Error("initConnection: If you use auto_login and "+
                             "authentication='login' then you also need to provide a password.");
                     }
-                    this.jid = Strophe.getBareJidFromJid(this.jid).toLowerCase()+'/'+Strophe.getResourceFromJid(this.jid);
+                    var resource = Strophe.getResourceFromJid(this.jid);
+                    if (!resource) {
+                        this.jid = this.jid.toLowerCase() + '/converse.js-' + Math.floor(Math.random()*139749825).toString();
```

Comment:
Can you please make a method `generateResource` (under the `Module-level functions` section) that generates the resource?

In other words, it contains the code: `/converse.js-' + Math.floor(Math.random()*139749825).toString();`

And then in the other place where the resource is generated that method should also be called.

Thanks